File: lib/android_build/maesdk/src/main/java/com/microsoft/applications/events/CommonDataContext.java
//
// Copyright (c) 2015-2020 Microsoft Corporation and Contributors.
// SPDX-License-Identifier: Apache-2.0
//
package com.microsoft.applications.events;

import java.util.Vector;

/**
 * `CommonDataContexts` provide a convenient struct to convey common information to
 * detect in the data being uploaded. This is expected to be internal information that
 * the host application may have access to, but do not wish to be uploaded via the 1DS
 * SDK path.
 */
public class CommonDataContext {
    /**
    * Domain Name for the current machine
    */
    public String domainName = "";

    /**
    * Friendly Machine Name
    */
    public String machineName = "";

    /**
     * List of multiple usernames for multi-user scenarios.
     */
    public Vector<String> userNames = new Vector<>();

    /**
     * List of multiple User Aliases for multi-user scenarios.
     */
    public Vector<String> userAliases = new Vector<>();
    /**
    * IP Addresses for local network ports such as IPv4, IPv6, etc.
    */
    public Vector<String> ipAddresses = new Vector<>();

    /**
    * Collection of Language identifiers
    */
    public Vector<String> languageIdentifiers = new Vector<>();

    /**
    * Collection of Machine ID such as Machine Name, Motherboard ID, MAC Address, etc.
    */
    public Vector<String> machineIds = new Vector<>();

    /**
    * Collection of OutOfScope Identifiers such as SQM_ID, Client_ID, etc.
    */
    public Vector<String> outOfScopeIdentifiers = new Vector<>();
}

File: wrappers/swift/Sources/OneDSSwift/CommonDataContext.swift
//
// Copyright (c) Microsoft Corporation. All rights reserved.
// SPDX-License-Identifier: Apache-2.0
//

import ObjCModule

/// Wrapper over ODWCommonDataContext class.
public final class CommonDataContext {
    ///
    public let odwCommonDataContext: ODWCommonDataContext

    /// Contructor initialized with ObjC wrapped object.
    public init(odwCommonDataContext: ODWCommonDataContext) {
        self.odwCommonDataContext = odwCommonDataContext
    }

    /// Domain Name for the current machine.
    public var domainName:String {
        get {
            odwCommonDataContext.domainName
        }
        set {
            odwCommonDataContext.domainName = newValue
        }
    }

    /// Friendly Machine Name
    public var machineName:String {
        get {
            odwCommonDataContext.machineName
        }
        set {
            odwCommonDataContext.machineName = newValue
        }
    }

    /// User Name such as the Login Name.
    public var userNames:[Any] {
        get {
            odwCommonDataContext.userNames as? [Any] ?? []
        }
        set {
            odwCommonDataContext.userNames = NSMutableArray(array: newValue)
        }
    }

    /// User Alias, if different that UserName.
    public var userAliases:[Any] {
        get {
            odwCommonDataContext.userAliases as? [Any] ?? []
        }
        set {
            odwCommonDataContext.userAliases = NSMutableArray(array: newValue)
        }
    }

    /// IP Addresses for local network ports such as IPv4, IPv6, etc.
    public var IPAddresses:[Any] {
        get {
            odwCommonDataContext.IPAddresses as? [Any] ?? []
        }
        set {
            odwCommonDataContext.IPAddresses = NSMutableArray(array: newValue)
        }
    }

    /// Collection of Language Identifiers.
    public var languageIdentifiers:[Any] {
        get {
            odwCommonDataContext.languageIdentifiers as? [Any] ?? []
        }
        set {
            odwCommonDataContext.languageIdentifiers = NSMutableArray(array: newValue)
        }
    }

    /// Collection of Machine Identifies such as Machine Name, Motherboard ID, MAC Address, etc.
    public var machineIDs:[Any] {
        get {
            odwCommonDataContext.machineIDs as? [Any] ?? []
        }
        set {
            odwCommonDataContext.machineIDs = NSMutableArray(array: newValue)
        }
    }

    /// Collection of Out-of-Scope identifiers such as Client ID, etc.
    public var outOfScopeIdentifiers:[Any] {
        get {
            odwCommonDataContext.outOfScopeIdentifiers as? [Any] ?? []
        }
        set {
            odwCommonDataContext.outOfScopeIdentifiers = NSMutableArray(array: newValue)
        }
    }
}
